### PR TITLE
Gate _jit_pass_onnx_eval_peephole on the do_constant_folding flag

### PR DESCRIPTION
**Pull request by geohot**
This optimization pass will merge BatchNorm into Conv, which is a form of constant folding.

This caused a regression for us from 1.6.0, where we could turn off constant folding and use the weights from the state_dict to evaluate the model. In 1.7.0, there's no way to turn this fusion off in eval mode.

cc @BowenBao @neginraoof

### CODE REVIEW

**Comment by garymm** [comment]
Hi @geohot could you please explain your use case a bit more? If you need to inspect the weights, Can you use `training=TrainingMode.PRESERVE` or `training=TrainingMode.TRAINING` to achieve your goal? If not, why not?

Also if we do decide to move forward with this change, please add a unit test to enforce the behavior.

**Comment by garymm** [comment]
Closing due to lack of response from @geohot. Please open another PR if needed.
